dmsite/frontend/src/Profile/ProfileComponent.jsx

```diff
@@ -1,20 +1,65 @@
 import React from 'react';
+import { FormErrors } from './FormErrors';
 import { withRouter } from 'react-router-dom';
 import {
-  Container, Card, Button, CardTitle, CardBody, InputGroup,
-  InputGroupAddon, Input, Popover, PopoverHeader, PopoverBody,
+  Container, Card, Button, CardBody, InputGroup,
+  Input, CardHeader,
 } from 'reactstrap';
 
 class Profile extends React.Component {
+
   constructor(props) {
     super(props);
-
     this.toggle = this.toggle.bind(this);
     this.state = {
       popoverOpen: false,
+      username: '',
+      email: '',
+      password: '',
+      formErrors: {email: '', password: '', username: ''},
+      usernameValid: false,
+      emailValid: false,
+      passwordValid: false,
+      formValid: false
     };
   }
 
+  handleUserInput = (e) => {
+    const name = e.target.name;
+    const value = e.target.value;
+    this.setState({[name]: value},
```

Comment:
What is going on here? Why do you need to set the value of name here?

Reply:
Then name and value are used to verify if the given input is valid